### PR TITLE
Adding onInitialTimestamp and changes for consistentSnapshot

### PR DESCRIPTION
**Pull request by anu3990**
https://github.com/marklogic/node-client-api/issues/622 - Added onInitialTimestamp and updated consistentSnapshot options for queryAll api
https://github.com/marklogic/node-client-api/issues/620 - Updated consistentSnapshot option for readAll api.

### CODE REVIEW

**Review by ehennum — CHANGES_REQUESTED** [review]
Feedback provided in email

**Review by ehennum — COMMENTED** [review]
This update seems headed in the right direction

File: lib/documents.js [hunk 1]
```diff
@@ -2538,8 +2538,14 @@ function queryAllDocumentsImpl(query, jobOptions) {
     serverTimestamp: null
   };
 
-  if(jobState.jobOptions.consistentSnapshot && typeof jobState.jobOptions.consistentSnapshot !== 'boolean'){
-    throw new Error('consistentSnapshot needs to be a boolean.');
+  if(jobState.jobOptions.consistentSnapshot){
+    if(typeof jobState.jobOptions.consistentSnapshot !== 'boolean' &&
+        (typeof jobState.jobOptions.consistentSnapshot !== 'object' || !jobState.jobOptions.consistentSnapshot.valueOf('Timestamp'))){
```
Comment (ehennum) [inline lib/documents.js:2543]: I'd suggest:

    typeof jobState.jobOptions.consistentSnapshot !== 'boolean &&
    !(jobState.jobOptions.consistentSnapshot instanceof Boolean) &&
    !(jobState.jobOptions.consistentSnapshotinstanceof instanceof mlutil.Timestamp)

JavaScript has both primitive and object versions of booleans and numbers (similar to Java), and the constructor of the Timestamp object can be used to verify the type similar to:

https://github.com/marklogic/node-client-api/blob/master/lib/documents.js#L1885

The same comment applies to the other initial validation.

File: lib/documents.js [hunk 2]
```diff
@@ -2581,7 +2587,11 @@ function onQueryUrisRequest(output) {
   jobState.startTime = Date.now();
   jobState.endpoint = endpoint;
 
-  if(jobState.jobOptions.consistentSnapshot){
+  if(jobState.jobOptions.consistentSnapshot && (typeof jobState.jobOptions.consistentSnapshot === 'object')){
```
Comment (ehennum) [inline lib/documents.js:2590]: A Boolean object would pass this test but not a test using instanceof mlutil.Timestamp

The same comment applies to the other initial validation.

**Review by ehennum — COMMENTED** [review]
Other than some suggestions about the conditional testing, this update looks good.

**Comment by anu3990** [comment]
Merging all the commits in a single commit after adding all the changes and suggestions.

**Pull request merged**